### PR TITLE
Update docs and add state for junos_l2_interfaces

### PR DESCRIPTION
**Pull request by danielmellado**
This commit adds gathered state for junos_l2_interfaces and updates the
documentation to the new resource modules versioning for collections.

### CODE REVIEW

**Comment by pabelanger** [comment]
recheck

File: plugins/modules/junos_l2_interfaces.py [hunk 1]
```diff
@@ -30,14 +30,13 @@
...
-
-DOCUMENTATION = """module: junos_l2_interfaces
-short_description: Manage Layer-2 interface on Juniper JUNOS devices
+ANSIBLE_METADATA = {"metadata_version": "1.1", "supported_by": "network"}
+
+DOCUMENTATION = """
+---
+module: junos_l2_interfaces
+version_added: "1.0.0"
+short_description: JUNOS L2 interfaces resource module.
```
Comment (samccann) [inline plugins/modules/junos_l2_interfaces.py:39]:
```suggestion
short_description: Junos L2 interfaces resource module
```

**Comment by danielmellado** [comment]
recheck

**Comment by danielmellado** [comment]
recheck

**Pull request merged**